lib/Dialect/FIRRTL/Transforms/LowerTypes.cpp

```diff
@@ -514,7 +541,8 @@ bool TypeLoweringVisitor::lowerArg(Operation *module, size_t argIndex,
   // Flatten any bundle types.
   SmallVector<FlatBundleFieldEntry> fieldTypes;
   auto srcType = newArgs[argIndex].type.cast<FIRRTLType>();
-  if (!peelType(srcType, fieldTypes, preserveAggregate))
+  if (!peelType(srcType, fieldTypes,
+                isModuleAllowedToPreserveAggregate(module)))
```

Comment:
Do you want to call peelTypes at all if not preserving aggregates?  Note that Aggregates is a general term that includes arrays and structs.  If you just mean structs, please rename to indicate that.

Reply:
I modified peelType implementation not to peel types if aggregate preservation is enabled and certain conditions hold.It might be better to separate the logic though. 

> Note that Aggregates is a general term that includes arrays and structs. If you just mean structs, please rename to indicate that.

I have handled arrays as well.